eslint.config.js

```diff
@@ -0,0 +1,249 @@
+          'auth',
+          'openApi',
+          '_cacheLength',
+          'defaultBadgeData',
+          'render',
+          'constructor',
+          'fetch',
+          'transform',
+          'handle',
+        ],
+      },
+    ],
+  },
+}
+
+// config specific to linting Mocha tests
+const mochaConfig = {
+  files: [
+    '**/*.spec.@(js|mjs|ts)',
+    '**/*.integration.js',
+    '**/test-helpers.js',
+    'core/service-test-runner/**/*.js',
+  ],
+
+  plugins: {
+    mocha: mochaPlugin,
+  },
+
+  languageOptions: {
+    globals: {
+      ...globals.mocha,
+    },
+  },
+
+  rules: {
+    'mocha/no-exclusive-tests': 'error',
+    'mocha/no-skipped-tests': 'error',
+    'mocha/no-mocha-arrows': 'error',
+    'mocha/prefer-arrow-callback': 'error',
+    'no-unused-expressions': 'off',
+  },
+}
+
+// config specific to linting Cypress tests
+const cypressConfig = {
+  files: ['**/*.cy.@(js|ts)'],
+  ...cypressPlugin.configs.recommended,
+}
+// append these to cypress.configs.recommended, without overwriting
+cypressConfig.plugins.mocha = mochaPlugin
+cypressConfig.rules['mocha/no-exclusive-tests'] = 'error'
+cypressConfig.rules['mocha/no-skipped-tests'] = 'error'
+cypressConfig.rules['mocha/no-mocha-arrows'] = 'off'
+
+// config specific to linting Service tests (IcedFrisby)
+const serviceTestsConfig = {
+  files: ['services/**/*.tester.js'],
+
+  plugins: {
+    icedfrisby: icedfrisbyPlugin,
+  },
+
+  rules: {
+    'icedfrisby/no-exclusive-tests': 'error',
+    'icedfrisby/no-skipped-tests': 'error',
+    'no-unused-expressions': 'off',
+  },
+}
+
+// config specific to linting JSDoc comments
+const jsDocConfig = {
+  plugins: {
+    jsdoc: jsdocPlugin,
+  },
+
+  rules: {
+    'jsdoc/require-jsdoc': 'off',
+    'jsdoc/no-undefined-types': ['error', { definedTypes: ['Joi'] }],
+    'jsdoc/check-alignment': 'error',
+    'jsdoc/check-param-names': 'error',
+    'jsdoc/check-tag-names': 'error',
+    'jsdoc/check-types': 'error',
+    'jsdoc/implements-on-classes': 'error',
+    'jsdoc/tag-lines': ['error', 'any', { startLines: 1 }],
+    'jsdoc/require-param': 'error',
+    'jsdoc/require-param-description': 'error',
+    'jsdoc/require-param-name': 'error',
+    'jsdoc/require-param-type': 'error',
+    'jsdoc/require-returns': 'error',
+    'jsdoc/require-returns-check': 'error',
+    'jsdoc/require-returns-description': 'error',
+    'jsdoc/require-returns-type': 'error',
+    'jsdoc/valid-types': 'error',
+  },
+}
+
+const config = [
+  {
+    ignores: [
+      'api-docs/',
+      'build',
+      'coverage',
+      '__snapshots__',
+      'public',
+      'badge-maker/node_modules/',
+      '!.github/',
+      'frontend/.docusaurus/**',
+    ],
+  },
+
+  js.configs.recommended,
+  chaiFriendlyPlugin.configs.recommendedFlat,
+  ...neostandard({ noStyle: true }),
+
+  globalConfig,
+  commonJsConfig,
+  nodeEsmConfig,
+  frontendConfig,
+  servicesConfig,
+  mochaConfig,
+  cypressConfig,
+  serviceTestsConfig,
+  jsDocConfig,
```

Comment:
The way I've structured this, I've written lots of small ESLint Config objects and then composed them all at the end here.

If we wanted, we could actually split these all out into individual files. I haven't done it here, but I am happy to do it if we think it would be useful.

Reply:
I'd say it's helpful to have all the config objects in one place, let's keep as is. 👍🏻